assets/shop/js/card_form.js

```diff
@@ -0,0 +1,271 @@
+import * as JSEncrypt from './jsencrypt.min';
+
+export class CardForm {
+  #form;
+  #cardHolderName;
+  #cardNumber;
+  #cardOperatorIcon;
+  #cardsApi;
+  #cvc;
+  #encryptedCard;
+  #expirationMonth;
+  #expirationYear;
+  #submitButton;
+
+  constructor(selector) {
+    this.#form = document.querySelector(selector);
+    this.#cardHolderName = this.#form.querySelector('[data-tpay-card-holder-name]');
+    this.#cardNumber = this.#form.querySelector('[data-tpay-card-number]');
+    this.#cardOperatorIcon = this.#form.querySelector('[data-tpay-card-operator-icon]');
+    this.#cardsApi = this.#form.querySelector('[data-tpay-cards-api]');
+    this.#cvc = this.#form.querySelector('[data-tpay-cvc]');
+    this.#encryptedCard = this.#form.querySelector('[data-tpay-encrypted-card]');
+    this.#expirationMonth = this.#form.querySelector('[data-tpay-expiration-month]');
+    this.#expirationYear = this.#form.querySelector('[data-tpay-expiration-year]');
+    this.#submitButton = this.#form.querySelector('[type="submit"]');
+    this.#registerEvents();
+    this.#registerFormatters();
+  }
+
+  #registerCvcFormatting() {
+    this.#cvc.addEventListener('keyup', (event) => {
+      const value = event.target.value.replace(/\s/g, '');
+
+      event.target.value = value.slice(0, 3);
+    });
+  }
+
+  #registerCardNumberFormatting() {
+    this.#cardNumber.addEventListener('keypress', (event) => {
+      const value = event.target.value.replace(/\s/g, '');
+
+      if (value.length >= 16) {
```

Comment:
I'm for keeping or integers (except some simple one like 0, 1) in consts. Then you can reuse it in lines 54, 60.

So for example:
```suggestion
      if (value.length >= CARD_NUMBER_LENGHT) {
```